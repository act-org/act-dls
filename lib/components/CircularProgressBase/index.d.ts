/**
 * @prettier
 */
import * as React from 'react';
import { CircularProgressProps } from '@material-ui/core';
export declare type Props = CircularProgressProps;
declare const CircularProgressBase: React.FC<Props>;
export default CircularProgressBase;
//# sourceMappingURL=index.d.ts.map