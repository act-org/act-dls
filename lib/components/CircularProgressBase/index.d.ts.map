{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/CircularProgressBase/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAC/B,OAAO,EAAoB,qBAAqB,EAAE,MAAM,mBAAmB,CAAC;AAE5E,oBAAY,KAAK,GAAG,qBAAqB,CAAC;AAE1C,QAAA,MAAM,oBAAoB,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAEmB,CAAC;AAE9D,eAAe,oBAAoB,CAAC"}