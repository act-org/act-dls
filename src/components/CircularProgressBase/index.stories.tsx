/**
 * @prettier
 */

import * as React from 'react';
import { select, text } from '@storybook/addon-knobs';

import CircularProgressBase from '.';

export const Base = (): React.ReactElement<any> => {
  const [value, setValue] = React.useState(0);

  React.useEffect((): (() => any) => {
    const interval = setInterval((): void => {
      setValue((v): number => (v >= 100 ? 0 : v + 1));
    }, 100);

    return (): any => clearInterval(interval);
  }, []);

  return (
    <CircularProgressBase
      size={Number(text('Size', '24'))}
      thickness={Number(text('Thickness', '4.5'))}
      value={value}
      variant={select(
        'Variant',
        {
          Determinate: 'determinate',
          Indeterminate: 'indeterminate',
          Static: 'static',
        },
        'indeterminate',
      )}
    />
  );
};

export default {
  component: CircularProgressBase,
  parameters: {
    info: {
      text: 'This is the base variant of the CircularProgress component.',
    },
  },
  title: 'Atoms/CircularProgress',
};
