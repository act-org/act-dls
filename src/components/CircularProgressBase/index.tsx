/**
 * @prettier
 */

import * as React from 'react';
import { CircularProgress, CircularProgressProps } from '@material-ui/core';

export type Props = CircularProgressProps;

const CircularProgressBase: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => <CircularProgress {...props} />;

export default CircularProgressBase;
