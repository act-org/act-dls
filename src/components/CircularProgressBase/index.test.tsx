/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import render from '~/helpers/test/render';

import CircularProgressBase from '.';

describe('CircularProgressBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <CircularProgressBase size={30} thickness={4} />,
    );

    expect(container).toMatchSnapshot();
  });
});
