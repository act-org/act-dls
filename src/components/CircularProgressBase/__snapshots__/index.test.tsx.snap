// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgressBase matches the snapshot 1`] = `
<div>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 30px; height: 30px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20"
        stroke-width="4"
      />
    </svg>
  </div>
</div>
`;
